vault-plugins/flant_iam/model/role.go

```diff
@@ -123,15 +134,7 @@ func (r *RoleRepository) Update(role *Role) error {
 	return r.save(role)
 }
 
-func (r *RoleRepository) Delete(id RoleName) error {
-	role, err := r.GetByID(id)
-	if err != nil {
-		return err
-	}
-	return r.db.Delete(RoleType, role)
-}
-
-func (r *RoleRepository) List() ([]*Role, error) {
+func (r *RoleRepository) List(showArchived bool) ([]*Role, error) {
```

Comment:
I would try not to break repo interfaces. I'd keep them similar and concise. Otherwise repo users will have to pass something instead of having good common case.  List argument can be parent ID, because sometimes it is just unavoidable.    

I'd make a use case and use Iter to handle filtering. And would not change the repo for it.

Reply:
I tried to make code faster. If we make special index for "archived" (I still think about it), it will make List with  "showArchived" faster then filter in use case. EveryWhere in List, our web application need to have possibility to pass   "showArchived", so it looks better to change signature in Repository.